src/modeler/AutoRest.Swagger/OperationBuilder.cs

```diff
@@ -243,17 +235,30 @@ private void BuildMethodParameters(Method method)
             foreach (var swaggerParameter in DeduplicateParameters(_operation.Parameters))
             {
                 var parameter = ((ParameterBuilder)swaggerParameter.GetBuilder(_swaggerModeler)).Build();
-                method.Add(parameter);
+                var actualSwaggerParameter = _swaggerModeler.Unwrap(swaggerParameter);
+
+                if (parameter.IsContentTypeHeader){
+                    // you have to specify the content type, even if the OpenAPI definition claims it's optional
+                    parameter.IsRequired = true;
+                    // enrich Content-Type header with "consumes"
+                    if (actualSwaggerParameter.Enum == null && 
+                        swaggerParameter.Extensions.GetValue<JObject>("x-ms-enum") == null &&
+                        _effectiveConsumes.Count > 1)
+                    {
+                        actualSwaggerParameter.Enum = _effectiveConsumes.ToList();
+                        actualSwaggerParameter.Extensions["x-ms-enum"] = 
+                            JObject.FromObject(new
+                            {
+                                name = "ContentType",
```

Comment:
So.... what if there are two APIs that take different content types? 

Both APIs will expect their `ContentType` enum class to be called "`ContentType`" ? (and end up in `models`?)